docs/installation/application.rst

```diff
@@ -0,0 +1,182 @@
+===========================
+
+You will need to copy the source code of the app to the server. There are many methods for accomplishing this, but one way is to create a repository for your code in GitHub. To download the source from GitHub, clone it as follows::
+
+    $ cd $TETHYS_HOME/apps/
+    $ sudo git clone https://github.com/username/tethysapp-my_first_app.git
+
+.. tip::
+
+    Substitute "username" for your GitHub username or organization and substitute "tethysapp-my_first_app" for the name of the repository with your app source code.
+
+2. Install the App
+==================
+
+Using Install (Recommended):
+-------------------------
+
+Using the ``install`` cli command, you can install the application in one quick step. The ``install`` command will install all the dependencies, link any services that you might need for your application and can also rely on the portal configuration to link default services to your application (as configured by the portal administrator).
+The install command uses three configuration files:
+
+.. _tethys_install_yml:
+
+install.yml 
+--------
+
+This file is generated with your application scaffold. Please do NOT list any dependencies in setup.py. Instead list them in the :file:`install.yml` file. This file should be committed with your application code in order to aid installation on a Tethys Portal
+
+.. code-block:: python
+
+	# This file may be committed to your app code.
+	version: 1.0
+	# This should match the app - package name in your setup.py
+	name: hydroexplorer
+
+	conda:
+	  # Putting in a skip true param will skip the entire section
+	  skip: false
+	  channels:
+	  	- tacaswell
+	  dependencies:
+	    - pyshp=2.0.0
+	    - geojson
+	    - shapely
+	  post:
+  	    - ./test.sh
+
+
+**install.yml Options:**
+
+* **version**: Indicated the version of the :file:`install.yml` file. Current default : 1.0
+* **name**: This should match the app-package name in your setup.py
+
+* **conda/skip**: If enabled, it will skip the installation of dependencies and channels
+
+* **conda/channels**: List of conda channels that need to be searched for dependency installation. Only specify any conda channels that are apart from the default. 
+
+* **conda/dependencies**: List of python dependencies that need to be installed by conda. These may be entered in the format ``pyshp=2.0.0`` to download a specific version. 
+
+* **Post**: A list of shell scripts that you would want to run after the installation is complete. This option can be used to initialize workspaces/databases etc. 
+
+.. _tethys_services_yml:
+
+
+services.yml 
+------------
+
+This file will be created by the portal administrator who has created/has access to all the service in the portal. This file will only be run by default if there is no portal services config file present (see below). However you can force the use of this file over the portal config by specifying the `--force-services` tag on the install command.  
```

Comment:
I think the default should be current behavior. It's installed with nothing configured. Tethys should recognize that it is not configured and only be visible to admins. The admin is directed to the settings page to configure the app.

Reply:
So no change is needed here?

Reply:
@tylorbayer do you know what happens when you run the install command and there is no services.yml or portal.yml? That's the behavior we're unsure of here. 

What we'd like it to do is launch into interactive configuration by default. However, we'd also like it to have an "--no-config" mode that would allow you to install an app without any configuration of settings or execution of extra scripts. If that's not the case, then we should change it.